File: andriod/src/CooperHybrid/src/com/codesharp/cooper/Repositories/TaskRepository.java
package com.codesharp.cooper.Repositories;

import java.util.List;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import com.codesharp.cooper.Constant;
import com.codesharp.cooper.MainActivity;
import com.codesharp.cooper.Task;
import com.codesharp.cooper.TaskDao;
import com.codesharp.cooper.Tasklist;
import com.codesharp.cooper.TasklistDao;

public class TaskRepository {
	private MainActivity _activity;
	private TaskDao _taskDao;
	private SharedPreferences _sharedPrefs;
	
	public TaskRepository(MainActivity activity) {
		this._activity = activity;
		this._taskDao = this._activity.getSession().getTaskDao();
		this._sharedPrefs = PreferenceManager.getDefaultSharedPreferences(this._activity.getApplicationContext());
	}
	
	public List<Task> getAllTaskByTemp()
	{
		List<Task> tasks = this._taskDao.queryBuilder()
				.where(TaskDao.Properties.AccountId.isNull())
				.orderAsc(TaskDao.Properties.TaskId)
				.list();
		return tasks;
	}

	public void updateTasklistIdByNewId(String oldId, String newId) {
		
		List<Task> tasks = this.getAllTask(oldId);
		for (Task task : tasks) {
			task.setTasklistId(newId);
		}
		if(tasks.size() > 0)
		{
			this._taskDao.updateInTx(tasks);
		}
	}

	public void updateTaskIdByNewId(String oldId, String newId, String tasklistId) {
		Task task = this.getTaskById(oldId);
		if(task != null)
		{
			task.setTaskId(newId);
		}
		this._taskDao.update(task);
	}

	public void deleteAll(String tasklistId) {
		
		if(this._sharedPrefs.contains(Constant.USERNAME_KEY))
		{
			String username = this._sharedPrefs.getString(Constant.USERNAME_KEY, "");
			this._taskDao.queryBuilder()
					.where(TaskDao.Properties.AccountId.eq(username)
							, TaskDao.Properties.TasklistId.eq(tasklistId))
					.buildDelete()
					.executeDeleteWithoutDetachingEntities();
		}
		else 
		{
			this._taskDao.queryBuilder()
					.where(TaskDao.Properties.AccountId.eq("")
							, TaskDao.Properties.TasklistId.eq(tasklistId))
					.buildDelete()
					.executeDeleteWithoutDetachingEntities();
		}
	}

	private List<Task> getAllTask(String tasklistId) {
		List<Task> tasks;
		if(this._sharedPrefs.contains(Constant.USERNAME_KEY))
		{
			String username = this._sharedPrefs.getString(Constant.USERNAME_KEY, "");
			tasks = this._taskDao.queryBuilder()
					.where(TaskDao.Properties.AccountId.eq(username)
							, TaskDao.Properties.TasklistId.eq(tasklistId))
					.orderAsc(TaskDao.Properties.TaskId)
					.list();
		}
		else 
		{
			tasks = this._taskDao.queryBuilder()
					.where(TaskDao.Properties.AccountId.eq("")
							, TaskDao.Properties.TasklistId.eq(tasklistId))
					.orderAsc(TaskDao.Properties.TaskId)
					.list();
		}
		return tasks;
	}

	public void addTasks(List<Task> tasks) {
		
		this._taskDao.insertInTx(tasks);
	}

	public Task getTaskById(String taskId) {
		List<Task>tasks = this._taskDao.queryBuilder()
			.where(TaskDao.Properties.TaskId.eq(taskId))
			.list();
		
		Task task = null;
		if(tasks.size() > 0)
		{
			task = tasks.get(0);
		}
		return task;
	}

	public void addTask(Task task) {
		this._taskDao.insert(task);
	}

	public void updateTask(Task task) {
		this._taskDao.update(task);
	}

	//批量更新任务
	public void updateTasks(List<Task> tasks) {
		
		this._taskDao.updateInTx(tasks);
	}

	public void deleteTask(Task task) {
		
		this._taskDao.delete(task);
	}
	
	
}

File: windows phone/src/Cooper.Repositories/TaskRepository.cs
using System;
using System.Collections.Generic;
using System.IO.IsolatedStorage;
using System.Linq;
using System.Net;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Documents;
using System.Windows.Ink;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Animation;
using System.Windows.Shapes;
using Cooper.Core;
using Cooper.Core.Models;

namespace Cooper.Repositories
{
    public class TaskRepository
    {
        private CooperContext _context;

        public TaskRepository()
        {
            this._context = new CooperContext();
        }

        public List<Task> GetAllTaskByTemp()
        {
            List<Task> tasks = this._context.Tasks
                .Where(o => o.AccountId.Equals(""))
                .ToList();
            return tasks;
        }

        public List<Task> GetAllTask(string tasklistId)
        {
            List<Task> tasks = null;
            if (IsolatedStorageSettings.ApplicationSettings.Contains(Constant.USERNAME_KEY))
            {
                string username = IsolatedStorageSettings.ApplicationSettings[Constant.USERNAME_KEY] as string;
                tasks = this._context.Tasks.Where(o => o.AccountId.Equals(username)
                    && o.TasklistId.Equals(tasklistId))
                    .OrderBy(o => o.TaskId)
                    .ToList();
            }
            else
            {
                tasks = this._context.Tasks.Where(o => o.AccountId.Equals("")
                    && o.TasklistId.Equals(tasklistId))
                    .OrderBy(o => o.TaskId)
                    .ToList();
            }
            return tasks;
        }

        public Task GetTaskById(string taskId)
        {
            List<Task> tasks = this._context.Tasks.Where(o => o.TaskId.Equals(taskId)).ToList();
            Task task = null;
            if (tasks.Count > 0)
            {
                task = tasks[0];
            }
            return task;
        }

        public void UpdateTasklistIdByNewId(string oldId, string newId)
        {
            List<Task> tasks = this.GetAllTask(oldId);
            foreach (var task in tasks)
            {
                task.TasklistId = newId;
            }
            if (tasks.Count > 0)
            {
                this._context.SubmitChanges();
            }
        }

        public void UpdateTaskIdByNewId(string oldId, string newId, string tasklistId)
        {
            Task task = this.GetTaskById(oldId);
            if (task != null)
            {
                task.TaskId = newId;
            }
            this._context.SubmitChanges();
        }

        public void DeleteAll(string tasklistId)
        {
            List<Task> tasks = null;
            if (IsolatedStorageSettings.ApplicationSettings.Contains(Constant.USERNAME_KEY))
            {
                string username = IsolatedStorageSettings.ApplicationSettings[Constant.USERNAME_KEY] as string;
                tasks = this._context.Tasks.Where(o => o.AccountId.Equals(username)
                    && o.TasklistId.Equals(tasklistId))
                    .ToList();
            }
            else
            {
                tasks = this._context.Tasks.Where(o => o.AccountId.Equals("")
                    && o.TasklistId.Equals(tasklistId))
                    .ToList();
            }
            if (tasks.Count > 0)
            {
                this._context.Tasks.DeleteAllOnSubmit<Task>(tasks);
                this._context.SubmitChanges();
            }
        }

        public void AddTasks(List<Task> tasks)
        {
            this._context.Tasks.InsertAllOnSubmit<Task>(tasks);
            this._context.SubmitChanges();
        }

        public void AddTask(Task task)
        {
            this._context.Tasks.InsertOnSubmit(task);
            this._context.SubmitChanges();
        }

        public void UpdateTask(Task task)
        {
            this._context.SubmitChanges();
        }

        public void UpdateTasks(List<Task> tasks)
        {
            this._context.SubmitChanges();
        }

        public void DeleteTask(Task task)
        {
            this._context.Tasks.DeleteOnSubmit(task);
            this._context.SubmitChanges();
        }
    }
}
